teps/0057-windows-support.md

```diff
@@ -0,0 +1,154 @@
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Conformance](#conformance)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+The current state of Windows containers requires that they be run on Windows nodes, it is not possible to run Windows containers on a Linux system. For use cases that require the use of Windows containers there is no choice but to run those workloads on Windows nodes. The current state of Tekton does not support Windows nodes in any form, and so there is currently no way to run Tekton Pipelines to support Windows workloads. The goal of this TEP is to provide support for Windows workloads within Tekton Pipelines, meaning that support for Windows nodes will be added. In addition to supporting Windows nodes within Tekton Pipelines we would also need to ensure that all Tekton components (the Operator, Triggers, the CLI/Dashboard and the Catalog/Hub) support, or reflect, Tasks which require Windows nodes.
+
+## Motivation
+Our specific use case is building Unreal Engine projects in containers. It is not possible to build UE projects for Windows on Linux, so we must be able to build in Windows containers, which in turn requires support for Windows nodes. There are other scenarios which require the use of Windows nodes/containers and this TEP is to support any Windows-based use case. There is currently no viable alternative approach available here (see alternatives below). If we have a TaskRun that requires a Windows container then it must run on a Windows node. Currently this is an unsupported use case, and the goal of this TEP is to address this need.
+
+### Goals
+- Allow Tekton TaskRuns to selectively run on Windows nodes
+- Allow a TaskRun to use nodeSelector to request this feature
+- Core Tekton components (Operator, CLI and Dashboard) must be able to recognise and communicate with Windows nodes in the same way they do Linux nodes
+- Tasks on the Catalog and Hub should be able to indicate the requirement for a Windows node to run it
+- All current (Linux-only) workflows should remain unchanged
+
+### Non-Goals
+- The ability to run core Tekton components on Windows nodes is not a goal of this TEP
+    - It is expected that users will have a mix of Linux and Windows nodes and that the core Tekton controller components will run on Linux nodes
+- Tekton will not be responsible for determining whether a Pod for a TaskRun should be created on a Windows node. This will be an opt-in feature determined via a nodeSelector.
+
```

Comment:
another possible non-goal to include is supporting PipelineResources (at least initially) - many of the images we build and publish as part of tekton pipelines (https://github.com/tektoncd/pipeline/tree/main/cmd) are just to support PipelineResources; I think it would be reasonable to not support these in windows tasks which would reduce the number of images we need to update (but maybe you need them anyway?)

Reply:
That's a fair point - we don't need PipelineResources for our use case, so if we're happy to leave this until later that works for us :)